kod/object/active/holder/nomoveon/battler/player/user.kod

```diff
@@ -3685,6 +3706,78 @@ messages:
       return;
    }
 
+   UserGetFromContainer(items_list=$, numbers_list=$)
+   {
+      local oSplit, iCan_hold, oitem, iCantGet, iNumber;
+      iCantGet = 0;
+      for oitem in items_list
+      {
+
+         iNumber = First(numbers_list);
+
+         if NOT Send(self,@UserGetSafetyChecks,#what=oitem)
+         {
+            return;
+         }
+
+         if IsClass(oitem,&NumberItem)
+         {
+            iCan_hold = Send(self,@GetNumberCanHold,#what=oitem);
+            if iCan_hold < iNumber
+            {
+               if iCan_hold = 0
+               {
+                  Send(self,@MsgSendUser,#message_rsc=user_disallow_get,
+                        #parm1=Send(oitem,@GetDef),#parm2=Send(oitem,@GetName));
+                  Send(self, @WaveSendUser, #what=self, 
+                        #wave_rsc = user_cant_pickup_item_wav_rsc);
+                  iCantGet = 1;
+                }
+               else
+               {
+                  oSplit = Send(oitem,@Split,#number=iCan_hold);
+                  Send(self,@MsgSendUser,#message_rsc=user_got_some,
+                     #parm1=Send(oitem,@GetName));
+                  Send(self,@NewHold,#what=oSplit);
+                  iCantGet = 1;
+               }
+            }
+            % iCan_hold > iNumber
+            else
+            {
+               oSplit = Send(oitem,@Split,#number=iNumber);
+               Send(self,@NewHold,#what=oSplit);
```

Comment:
We used to check ReqNewOwner again here (any any time before @NewHold) but ReqNewOwner is already checked in the safety checks.  This redundancy exists in UserGet as well.  We can add back in or also remove from UserGet as appropriate.